baybe/utils/basic.py

```diff
@@ -11,6 +11,11 @@
 _U = TypeVar("_U")
 
 
+class _TClassWithSubclasses(Protocol):
+    def __subclasses__(self) -> List:
```

Comment:
ive also tried 
```
def __subclasses__(self: _T) -> List[_T]:
```
which works, but somehow seems incorrect to me. 

Instead more correct to me seems:
```
def __subclasses__(self: _T) -> List[Type[_T]]:
```
but that doesnt work so I went for the version without any `_T`

Reply:
Hi @Scienfitz, the issue is a bit subtle, but your protocol is not need:

The issue is that, apparently, generic types defined via TypeVars be default seem to be bound to **instances**, not classes. `__subclasses__`, however, is a classmethod, not an instance method. So with your protocol, you basically said that the **object** needs to support a `__subclasses__` method. This would not be needed if the typevar would refer to the class instead, which automatically has that method, because all classes have it (hope this is not too confusing). 

Thus, possible fixes:
1. mypy would probably no longer complain if you correctly turned your `__subclasses__` method in the protocol into a classmethod. Then your originally attempted could even work. But that would be a useless protocol, since classes support it anyway.
2. Realistic option A: put type(_T) on both input and output of `get_subclasses`
3. Probably more elegant version B: I introduced another typevar that is bound to `type` from the start. Works.

Have pushed the corresponding commit